diff -Nur gtk+-3.14.15.bak/util/Makefile.am gtk+-3.14.15/util/Makefile.am
--- gtk+-3.14.15.bak/util/Makefile.am	2015-07-21 23:22:37.000000000 +0900
+++ gtk+-3.14.15/util/Makefile.am	2016-07-22 16:21:49.529560650 +0900
@@ -1,9 +1,11 @@
 # The extract_strings tool is a build utility that runs on the build system.
 
-noinst_PROGRAMS = extract-strings
-
 extract_strings_SOURCES = extract-strings.c
 extract_strings_CFLAGS = $(GLIB_CFLAGS_FOR_BUILD)
 extract_strings_LDADD = $(GLIB_LIBS_FOR_BUILD)
+extract-strings$(BUILD_EXEEXT): $(extract_strings_SOURCES)
+	@rm -f extract-strings$(BUILD_EXEEXT)
+	$(AM_V_CCLD)$(CC_FOR_BUILD) $(extract_strings_CPPFLAGS) $(CPPFLAGS_FOR_BUILD) $(extract_strings_CFLAGS) $(CFLAGS_FOR_BUILD) $^ $(LDFLAGS_FOR_BUILD) $(extract_strings_LDADD) $(LIBS_FOR_BUILD) -o $@
+all-local: extract-strings$(BUILD_EXEEXT)
 
 -include $(top_srcdir)/git.mk
